### DISCUSSION TITLE
Proposal: gitea-releases

### DISCUSSION THREAD

**Discussion by justusbunsi** [renovatebot/renovate#20740] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

👋 Hi. First of all: Renovate is awesome. Thanks for maintaining this project.

I would like to propose a new datasource: `gitea-releases`. Similar to the `github-releases` it would be great to extract details from releases hosted on Gitea instances.

**Configuration**

Since Gitea is not one single instance but many self-hosted ones, the host should be part of the dependency name. A better alternative might be the "registry" concept from Docker.

**API**

There are only small differences between Gitea and GitHub. It _should_ be straight forward.

**Authentication**

I am not sure if access tokens are required for it to work in an initial state. Gitea doesn't have built-in rate limits, afaik.

For an actual authentication the concept from Docker/NPM registries with "Host -> Token" mapping could work.

**Comment by justusbunsi** [comment]
Not sure if I did something wrong with the proposal as I see down votes by maintainers. Was this discussed and maybe rejected somewhere else and I did not find it? Would it be too complex to maintain?

I appreciate any feedback so that I can understand the reactions and adjust the proposal if something is missing.

**Comment by rarkins** [comment]
![image](https://user-images.githubusercontent.com/6311784/222847623-1587fed3-3186-47df-8d53-a118467e4930.jpeg)

**Reply by justusbunsi** [reply]
Oh. Looks like this went down the wrong way. As of writing I am still unsure if I am allowed to sign the CLA, with regards to my job. In any other circumstance I would sign the CLA right away. It has absolutely nothing to do with the fact _that_ there is a CLA.

That's why I added that sentence in the first place. But reviewing that, I got your point and I understand that this is not a useful addition.

My apologies. I've removed that part.

**Comment by viceice** [comment]
this datasource should be modeled like the Gitlab releases.

so auth is already handled by lower http layer.

for it to work we only need the datasource, wich does the JSON request via http layer and transforms the results to renovate releases.

should be easy to implement.

**Comment by viceice** [comment]
Here is the `gitea-tags` datasource, the `gitea-releases` datasource will follow.
- #24083

**Reply by viceice** [reply]
- #24140

**Reply by viceice** [reply]
- #24144
